pyomo/contrib/doe/doe.py

```diff
@@ -991,179 +776,140 @@ def _create_doe_model(self, no_obj=True):
-        
-        # extract (discretized) time 
-        time_set=list(time_set_attr)
-        self.time_set = time_set
-
-        # create parameter, measurement, time and measurement time set
-        m.para_set = pyo.Set(initialize=list(self.param.keys()))
-        param_name = list(self.param.keys())
-        m.y_set = pyo.Set(initialize=self.jac_involved_name)
-        m.t_set = pyo.Set(initialize=time_set)
-
-        m.tmea_set = pyo.Set(initialize=self.timepoint_overall_set)
-
-        # we can be sure about the name of scenarios, because they are generated by our function
-        m.scenario = pyo.Set(initialize=scenario_all['scena-name'])
-        m.optimize = self.optimize
-
-        # check if measurement time points are in the time set
-        for j in m.y_set:
-            for t in m.tmea_set:
-                if not (t in time_set_attr):
-                    raise ValueError('Measure timepoints should be in the time list.')
-
-        # check if control time points are in the time set
-        for d in range(len(self.design_name)):
-            if self.design_time[d]:
-                for t in self.design_time[d]:
-                    if not (t in time_set_attr):
-                        raise ValueError('Control timepoints should be in the time list.')
-
-        ### Define variables
-        # Elements in Jacobian matrix
-        if self.jac_initial:
-            dict_jac = {}
-            for i, bu in enumerate(m.y_set):
-                for j, un in enumerate(m.para_set):
-                    for t, tim in enumerate(m.tmea_set):
-                        dict_jac[(bu,un,tim)] = self.jac_initial[i,j,t]
-
-            def jac_initialize(m,i,j,t):
-                return dict_jac[(bu,un,tim)]
-
-            m.jac = pyo.Var(m.y_set, m.para_set, m.tmea_set, initialize=jac_initialize)
+        # force design variables in blocks to be equal to global design values
+        for name in self.design_name:
+            
+            def fix1(mod, s):
+                cuid = pyo.ComponentUID(name)
+                design_var_global = cuid.find_component_on(mod)
+                design_var_global = cuid.find_component_on(mod)
+                design_var = cuid.find_component_on(mod.block[s])
+                return design_var == design_var_global
+            
+            con_name = "con"+name
+            mod.add_component(con_name, pyo.Constraint(mod.scena, expr=fix1)) 
 
-        else:
-            m.jac = pyo.Var(m.y_set, m.para_set, m.tmea_set, initialize=1E-20)
+        # variables for jacobian and FIM
+        mod.param = pyo.Set(initialize=list(self.param.keys()))
+        mod.res = pyo.Set(initialize=self.measure_name)
 
-        # Initialize Hessian with an identity matrix
-        def identity_matrix(m,j,d):
-            if j==d:
+        def identity_matrix(m,i,j):
+            if i==j:
                 return 1 
-            else: 
-                return 0
-
-        # initialize FIM
-        if self.fim_initial:
```

Comment:
Is this removing the ability to initialize the FIM with a non-identity matrix, e.g., computed with k-aug? Isn't this an important feature?

Reply:
added back.